src/main/java/net/logstash/logback/appender/AbstractLogstashTcpSocketAppender.java

```diff
@@ -1044,11 +1069,15 @@ public synchronized void stop() {
     }
 
     protected void fireEventSent(Socket socket, Event event, long durationInNanos) {
-        safelyFireEvent(l -> l.eventSent(this, socket, event, durationInNanos));
+        if (event != null) {
+            safelyFireEvent(l -> l.eventSent(this, socket, event, durationInNanos));
+        }
     }
 
     protected void fireEventSendFailure(Event event, Throwable reason) {
-        safelyFireEvent(l -> l.eventSendFailure(this, event, reason));
+        if (event != null) {
+            safelyFireEvent(l -> l.eventSendFailure(this, event, reason));
+        }
```

Comment:
In what case is event null now?

Reply:
When failed to send a keep alive... I moved the "if null" statement from the caller down inside the method itself to simplify calling code.